### ISSUE TITLE
All entity factories should use EntityNameResolver

### ISSUE THREAD

**Issue by TomasLudvik** [shopsys/shopsys#622]
Resolving entity in factories is great, but not all Factories are doing that, for example `GoogleFeedItemFactory`.

### Expected result
All factories are using EntityNameResolver instead of creating specific Entity.

**Comment by boris-brtan** [comment]
Here is bash search command that should find all problematic factory classes
```
find ./packages/framework/  -name '*Factory.php' ! -name '*DataFactory*'  -exec grep -Li "EntityNameResolver" {} \+
```

**Comment by PetrHeinz** [comment]
@boris-brtan not all of these are entity factories, but it is a helpful list nonetheless.

**Comment by vitek-rostislav** [comment]
Could we create an architectonical sniff for that as well?

**Comment by TomasLudvik** [comment]
Hello @vitek-rostislav,

this is something that should ease extendibility of Model. Can you please add this issue to some sprint in near future?

**Comment by vitek-rostislav** [comment]
resolved by #918
